### DISCUSSION TITLE
Installation and configuration help

### DISCUSSION THREAD

**Discussion by cos000** [johanmeijer/grott#136] [Q&A]
Hello
My system is composed of Growatt SPH6000 inverter with Shine Link datalogger (I can set growatt server on it like on shine wifi).
I successfully managed to install grott on a raspberry Pi 2.
Now I am a little bit confused:
What I need is an easy system to look at PV data via web page (best without login) on my home network and on the touch panels of my domotic system (they are such proprietary Android tablets). On them I can add custom links to web pages.
So what would be the best system?
MQTT and Grafana? But how does Grafana read MQTT? Do I need to install an extra plugin or is it better to use InfluxDB and then send data to Grafana?

Thank you so much

**Comment by ZonPV** [comment]
Easy system? Best system? Everybody will tell a different story. Depends on so many possibilities...
You don't tell what hardware and OS you use to run Grott... And your time and knowledge..

It is not difficult to configure Grott to send data to Influxdb or PVoutput. 
Influxdb is a good local database to store your data and present it to Grafana. But it needs knowledge to configure in your system. You have to innstall and configure Influxdb and Grafana
PVoutput will create nice graphics for you. Grott will send data to internet and PVoutpyt stores data in the cloud. Easy to configure in Grott and reach your goal... 
MQTT is just a way to transport your data to a database. You have to install software (broker ) for that too. 

Hopefully this makes it a bit clear?

**Comment by johanmeijer** [comment]
It all depends what you want. As @ZonPV says above mqtt is only a transport to another application that can process (store, report) the data. So you always need something in addition to that. 

Do you want to store historical information or only view the actual information? 

I have some solutions running (just for fun and experimenting). 

I use Nodered for reading the MQTT message and create actual reports (so no historical data). (I use Nodered also for others things but lets not make it to complex for now)

I use domoticz for home automation (others use Home Assistent). They are capable of handling MQTT messages from Grott and creating reports (and store historical data) on it. With home assistent you can process the grott MQTT message directly (see examples in the issues/discussions) for Domoticz I use ndered as a broker to create a MQTT message that can be handled by Domoticz. 

If you do not want to run your own applications Grott is capable of sending data to PVoutput.org (no mqtt needed). PVoutput will store and process the data and create reports (looks very nice, but only can process the data once per 5 minutes). 

If you want the complete thing (Store your own historical data and create customized reports) I think a database solutions with Grafana is the best. Grott can write directly to influxdb (no mqtt needed) and from there it is easy to setup reporting with Grafana (also examples in the discussion/issues). I also tested with other databases (Postgres and MongoDB) but needed some brokering for that in Nodered. 

I run everything at a RPI. Now I run on a RPI4 (with SSD)  in the past on a RPI2. I slowly moving to docker because it more easy eco system to maintain. So domoticz, Nodered, Grafana (and influxDB2 for testing) are running in a Docker Container. 
I also have a influxdb V1 running stand alone (at this moment my main database). 

My main Grott installation is still running stand alone (at the same RPI4) but I have also docker instances running on a x86 Ubuntu for testing (both Grott and Grottserver). 

So you have a lot of options and have to choose (I know this will cause some kind of choise stress, one of the biggest problems of our world at this moment). 

With your RPI2 (and if you use no Home automation tooling) I would run Grott, InluxDB (v1) and Grafana standalone (no docker). It is easy to install and maintain. The weak point of this installation is the SD-Card of the RPI. I had already multiple crashes with that and you have to do something to backup your influx data.  

Using MQTT and installing and do some experiments with Nodered does not hurts (and is really fun, but I am probably a Nerd) but is not necessary.

**Reply by cos000** [reply]
Hello

Thank you so much
I'd like to store my data and to view simple infos (no graphs but numbers like battery percentage, power form/to the battery ecc) for the moment. One day I may like to implement the system.
What is the main difference between docker and classic software?
Does docker system need better hardware?
I'd like to run this on a Raspberry. I have a 2 and a 4 which I can use.
Why InfluxDB v1 and not v2?
Should I use Raspberry Pi OS 64? Lite should be right?

Thank you

**Reply by ZonPV** [reply]
@cos000 
First you have to think how to store your data. 
In what format? Do you like to read a csv-file? Or in InfluxDB? To create graphics? I guess you want to store that data locally. More easy to handle.

Classic software will install straight into your operating system. 
Docker creates virtual machine(s) for (most of the time) a single task / application. It is easier to maintain or upgrade your application. Less conflicts between applications or software on the host.  It does not need better hardware. Even a Pi 2 will do for running Docker and Grottsoftware. If you like highspeed, take the Pi 4.

InfluxDB v1 is sufficient for your goal. Lightweight and good enough for local storage.

Raspberry Pi 64 bit is ok. The lite version can give surprises when you install applications later on. And a graphical interface is handy, once in a while. Maybe some extra costs for a larger SD-card but no problem I hope :)

My advise is to read the Wiki and install a broker, Mosquitto, Docker and virtual software for containermanagement. Like https://docs.portainer.io/v/ce-2.9/start/install. 
Than you can download from https://hub.docker.com/u/ledidobe the virtual machine for Grottsoftware of your choice 
and configure the ini-file of the app and that is it. 

Best regards

**Reply by johanmeijer** [reply]
In addition to @ZonPV

There is a CSV extension for Grott available: https://github.com/johanmeijer/grott/wiki/Extensions. That is probably the easiest way to collect data (it creates a CSV file per day). 

I advised influxDB1 because that is the one that can be installed easily with APT on a RPI2. Next to it in the beginning there were some issues with inlfuxdb v2. When you are going the use docker the influxdb v2 might be a go idea. 

I would not run docker on a RPI2. That will not perform very well. 

Docker helps you with easy installation and management of software., but ... You have to learn docker. Docker is not very difficult but if you are going to install databases etc  you have to find your way in where to put the data, how to backup etc. 

RPI2 => use the natively installed (APT) version of the software 
RPI4 => Go for docker and a management tool like Portainer as stated above.

**Comment by cos000** [comment]
Hello
Thank you for the replies.
I had no spare time so I had to put this project aside.
Last days I was trying to do something but without big results.

Fisrt of all I choosed to use RPI2 with Raspberry OS 32 bit (64 doesn’t run on RPI2) with a 64GB micro-SD card.
Once I got something working maybe I’ll try docker system on the RPI4.
So I installed the system, grott, influxDB 1.8 and grafana.

First question: can I test everything without being connected to the inverter? Should I see a flat 0 graphic but without errors? Just to see if the software talk to each other.

So, after I installed everything, I enabled the services, changed grott settings file to influxdb output, created the user in influxdb, gave it the permission and associated it to the grottdb, set influxdb datasource into grafana but I get error while trying to connect to the database.

The account data I set into grott.ini, into influxdb and into grafana datasource has to be the same right?

Thank you so much

**Comment by cos000** [comment]
So, I decided to move on docker because I like pain...

I made a new SDCard with RASPIOS 64.

I installed docker and portainer

I created a new volume for influxdb
I created a new container with:
image used= influxdb:latest
ports 8086 and 8088
volume mapping= var/lib/influxdb to the volume I created before.
restart policy= always

I created an admin and an another user in influxdb (same organization)
Created API token and config profile for the token...

Now I'm stuck

First of all: is it right what I have done since now? Do I need to do something else?

Is here someone who can tell me how to continue?

Thank you so much

**Reply by cliver1956** [reply]
Hi @cos000 Did you get anywhere with this? I have a similar problem in that I am running grott in a docker container on my NAS. It appears to be running OK as I have changed my growatt settings to point to it and the server.growatt.com is still getting data. However there is no data showing in my InfluxDB 2.4 bucket running in another container on the same NAS.
I don't want to steal your thread here but as there is some similarity I thought I'd link to it here in case it gives you a clue and someone may be able help me find out why I'm not writing to the database?
[https://github.com/johanmeijer/grott/discussions/193](url)

**Reply by johanmeijer** [reply]
@cos000 @cliver1956 Do you have grott logs from the startup (influx initialization) and the moment the data is (should be) sent to influxDB?
